src/app/dataset-view/additional-components/flows-component/components/flows-table/flows-table.component.ts

```diff
@@ -137,22 +154,79 @@ export class FlowsTableComponent implements OnInit, OnChanges {
                 noButtonText: "Cancel",
                 handler: (ok) => {
                     if (ok) {
-                        this.cancelFlowChange.emit(flowId);
+                        this.cancelFlowChange.emit({ flowId, datasetId });
                     }
                 },
             }),
         );
     }
 
-    public navigateToFlowDetaisView(flowId: string): void {
+    public navigateToFlowDetaisView(flow: FlowSummaryDataFragment, datasetId: string): void {
         this.navigationService.navigateToFlowDetails({
-            accountName: this.datasetBasics.owner.accountName,
-            datasetName: this.datasetBasics.name,
-            flowId,
+            accountName: this.datasetById(datasetId).owner.accountName,
+            datasetName: this.datasetById(datasetId).name,
+            flowId: flow.flowId,
         });
     }
 
     public dynamicImgSrc(status: FlowStatus): string {
         return DatasetFlowDetailsHelpers.dynamicImgSrc(status);
     }
+
+    public accountFormatter(x: Account | string): string {
+        return typeof x !== "string" ? x.accountName : x;
+    }
+
+    public searchByAccountName: OperatorFunction<string, readonly Account[]> = (text$: Observable<string>) =>
+        text$.pipe(
+            debounceTime(200),
+            distinctUntilChanged(),
+            tap(() => (this.searchingAccount = true)),
+            map((term) =>
+                (term === "" || !this.accountFlowInitiators
+                    ? []
+                    : this.accountFlowInitiators.filter(
+                          (initiator) => initiator.accountName.toLowerCase().indexOf(term.toLowerCase()) > -1,
+                      )
+                ).slice(0, 10),
+            ),
+            tap(() => (this.searchingAccount = false)),
+        );
+
+    public searchByDatasetName: OperatorFunction<string, readonly Dataset[]> = (text$: Observable<string>) =>
+        text$.pipe(
+            debounceTime(200),
+            distinctUntilChanged(),
+            tap(() => (this.searchingDataset = true)),
+            map((term) =>
+                (term === "" || !this.datasetFlowOwners
+                    ? []
+                    : this.datasetFlowOwners.filter(
+                          (initiator) => initiator.name.toLowerCase().indexOf(term.toLowerCase()) > -1,
+                      )
+                ).slice(0, 10),
+            ),
+            tap((result) => (this.searchAccountDatasetsLength = result.length)),
+            tap(() => (this.searchingDataset = false)),
+        );
```

Comment:
Make a service from these algorithms and unit tests heavily separately from component.

Reply:
Made small changes